src/Libraries/Microsoft.Extensions.AI/ChatCompletion/DistributedCachingChatClient.cs

```diff
@@ -99,14 +93,7 @@ protected override async Task WriteCacheStreamingAsync(string key, IReadOnlyList
     }
 
     /// <inheritdoc />
-    protected override string GetCacheKey(bool streaming, IList<ChatMessage> chatMessages, ChatOptions? options) =>
-        GetCacheKey([streaming ? _boxedTrue : _boxedFalse, chatMessages, options]);
-
-    /// <summary>Gets a cache key based on the supplied values.</summary>
-    /// <param name="values">The values to inform the key.</param>
-    /// <returns>The computed key.</returns>
-    /// <remarks>This provides the default implementation for <see cref="GetCacheKey(bool, IList{ChatMessage}, ChatOptions?)"/>.</remarks>
-    protected string GetCacheKey(ReadOnlySpan<object?> values)
+    protected override string GetCacheKey(params ReadOnlySpan<object?> values)
```

Comment:
The default implementation in `DistributedCachingChatClient` relies on the supplied objects being json serializable. Should we clarify this via comments somehow?